File: apple/Sources/ReactNativeWatchOS/Modifiers/BackgroundExtensionEffect.swift
import SwiftUI

/// SwiftUI `.backgroundExtensionEffect()` — watchOS 26 "Liquid Glass".
/// Extends the view's content beneath adjacent Liquid Glass surfaces
/// (mirrored/blurred) so the glass samples the underlying color. Takes no
/// parameters.
///
/// Gated to watchOS 26+ — returns the view unchanged on older OS.
enum RNWBackgroundExtensionEffectModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("backgroundExtensionEffect") { view, _, _ in
            guard #available(watchOS 26.0, *) else { return view }
            return AnyView(view.backgroundExtensionEffect())
        }
    }
}

File: src/modifiers/backgroundExtensionEffect.ts
import { createModifier } from './createModifier';

/// SwiftUI `.backgroundExtensionEffect()` (watchOS 26 "Liquid Glass").
/// Extends the view's content beneath adjacent Liquid Glass surfaces by
/// mirroring/blurring it, so the glass picks up the underlying color.
/// Takes no parameters. No-op on watchOS < 26.
export function backgroundExtensionEffect() {
  return createModifier('backgroundExtensionEffect');
}
